x-pack/plugins/canvas/canvas_plugin_src/elements/types.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export interface ElementSpec {
+  name: string;
+  image: string;
+  expression: string;
+  displayName?: string;
+  help?: string;
```

Comment:
`help` isn't strictly required, but the intention is that users should always include it. I assume the type should match the code (like it does now) and not the intention (ie. making the property not optional here) though, right? Just trying to understand the focus of types, this isn't a request for a change.

Reply:
If you really want to force the user to always supply the `help` parameter, maybe the code should throw an error if it's missing?  In which case, then code & intention & types can all match (e.g. not optional but required)?  Or is it more like it's "strongly encouraged" to add help, but not required?  In which case then I think type should indicate optional.

Reply:
> Or is it more like it's "strongly encouraged" to add help, but not required?

Yeah, that's the situation. You don't have to include it, but you get some default text (I think it might be empty, or maybe something like "help not available") if you don't.

Reply:
Personally, I think making it required is a good idea.  Having plugin developers intentionally put in an empty string if it has no help makes the API more obvious.  Having it be optional might guarantee no one ever fills it in, (to use @rashidkpc's term, 'copy/paste developers' might miss the help text entry entirely.